style/swift.md

```diff
@@ -1,3 +1,1037 @@
+  var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+}
+```
+
+* Long lines should be wrapped at around 170 characters. This is just a reference. A hard limit is intentionally not specified. 
+**Note:** Current limit defined in swiftlint is bigger than 170.
+
+* Avoid trailing whitespaces at the ends of lines.
+
+* Add a single newline character at the end of each file.
+
+## Availability qualifiers
+
+Use `@available(*, unavailable)` when using unimplemented init-with-coder initialisers.
+
+## Comments
+
+In the very rare occasions when they are needed, use comments to explain **why** a particular piece of code does something. Comments must be kept up-to-date, or altogether deleted. Never write comments that tell what the code **does**.
+
+Avoid block comments inline with code, as the code should be as self-documenting as possible. _Exception: This does not apply to those comments used to generate documentation._
+
+Avoid the use of C-style comments (`/* ... */`). Prefer the use of double- or triple-slash.
+
+## Classes and Structures
+
+### Which one to use?
+
+Remember, structs have [value semantics](https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/ClassesAndStructures.html#//apple_ref/doc/uid/TP40014097-CH13-XID_144). Use structs for things that do not have an identity. An array that contains `[a, b, c]` is really the same as another array that contains `[a, b, c]` and they are completely interchangeable. It doesn't matter whether you use the first array or the second, because they represent the exact same thing. That's why arrays are structs.
+
+Classes have [reference semantics](https://developer.apple.com/library/mac/documentation/Swift/Conceptual/Swift_Programming_Language/ClassesAndStructures.html#//apple_ref/doc/uid/TP40014097-CH13-XID_145). Use classes for things that do have an identity or a specific life cycle. You would model a ViewModel as a class because two ViewModel objects might be in a different state. Although they might represent the same screen, it doesn't mean they have the same state (e.g. `.loading` vs `.loaded([T])`.
+
+### Use of Self
+
+For conciseness, avoid using `self` since Swift does not require it to access an object's properties or invoke its methods.
+
+Use self only when required by the compiler (in `@escaping` closures, or in initializers to disambiguate properties from arguments). In other words, if it compiles without `self` then omit it.
+
+
+### Computed Properties
+
+For conciseness, if a computed property is read-only, omit the get clause. The get clause is required only when a set clause is provided. For single-expressions where the context is clear, both explicit and implicit returns can be used:
+
+**Preferred**:
+```swift
+var diameter: Double { radius * 2 }
+
+// Or
+
+var diameter: Double {
+  return radius * 2
+}
+```
+
+**Not Preferred**:
+```swift
+var diameter: Double {
+  get {
+    return radius * 2
+  }
+}
+```
+
+### Final
```

Comment:
We can add to this, or refine it to be more like [this](https://github.com/bbc/news-app-ios/blob/dev/iOSCodingGuide.md#final-classes)